server/region_syncer.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package server
+
+import (
+	"context"
+	"net/url"
+	"sync"
+
+	"github.com/pingcap/kvproto/pkg/pdpb"
+	log "github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+)
+
+type regionSyncer struct {
+	sync.Mutex
+	streams map[string]pdpb.PD_SyncRegionsServer
```

Comment:
I think streams can be managed in `Server` directly. As it is used for leader to sync regions to followers, while other fields are for sync regions with leader.